pages/goals.tsx

```diff
@@ -5,20 +5,44 @@ import supabase from '@/app/supabaseclient';
 import Goals from '@/app/components/goals';
 import Layout from '@/app/components/layout';
 import BasicModal from '@/app/components/goalmodal';
+import React from 'react';
+
 async function signOutUser() {
   const { error } = await supabase.auth.signOut();
   if (error) console.log('Error logging out:', error.message);
 }
 
 export default function Home() {
+  // I was intending on keeping high-level state here // State changing functions
+  // Modal state
+  const [modalState, setModalState] = React.useState(false);
+  const openModal = () => setModalState(true);
+  const closeModal = () => setModalState(false);
+
+  // Goals state
+  const [goalsList, setGoalsList] = React.useState([
+    { goal_title: 'Make a house', deadline: 'Tue Jan 19 2024 20:42:24`', description: '', priority: '', created_timestamp: `Tue Jan 09 2024 20:42:24`  },
+    { goal_title: 'Build a zoo', deadline: 'Tue Jan 29 2024 20:42:24`', description: '', priority: '', created_timestamp: `Tue Jan 09 2024 20:42:24` },
+    { goal_title: 'Start a zoom call', deadline: 'Tue Feb 03 2024 20:42:24`', description: '', priority: '', created_timestamp:`Tue Jan 09 2024 20:42:24` }
+  ]);
+
+  
+  const appendGoalsList = (goal) => { 
```

Comment:
![Screenshot from 2024-01-10 12-08-49](https://github.com/masterismail/achieve.ai/assets/111173270/fc5c5b0f-dc74-4329-aa26-a3dec0070803)

error log on vercel.

Reply:
Ah, it doesn't like that I didn't assign type to those. I'll work on that later tonight